cognite/client/_api/transformations/__init__.py

```diff
@@ -57,6 +57,16 @@ def create(
                 >>> res = c.transformations.create(transformations)
         """
         utils._auxiliary.assert_type(transformation, "transformation", [Transformation, list])
+
+        if isinstance(transformation, TransformationList):
+            sessions = {}
+            for t in transformation:
+                t._cognite_client = self._cognite_client
+                t._process_credentials(sessions)
+        elif isinstance(transformation, Transformation):
+            transformation._cognite_client = self._cognite_client
+            transformation._process_credentials({})
```

Comment:
I couldn't understand why we procecess empty dictionary when the item is not a list. :'( could you explain?

Reply:
the credentials that are processed are the ones inside the transformation, and the dictionary acts as a cache, to create only one session for repeated credentials.... maybe it would be clearer to just add a `_process_credentials()` method to TransformationList and change all this for a single line 😅 

```
transformation._process_credentials()
```